Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -32,9 +32,47 @@ public struct Configuration: Equatable {
     public let rules: [Rule]
     public var rootPath: String?              // the root path to search for nested configurations
     public var configurationPath: String?     // if successfully loaded from a path
-    public var hash: Int?
     public let cachePath: String?
 
+    internal var cacheDescription: String {
+        let cacheRulesDescriptions: [String: Any] = rules.reduce([:]) { accu, element in
+            var accu = accu
+            accu[type(of: element).description.identifier] = element.cacheDescription
+            return accu
+        }
+        let dict: [String: Any] = [
+            "root": rootPath ?? FileManager.default.currentDirectoryPath,
+            "rules": cacheRulesDescriptions
+        ]
+        if let jsonData = try? JSONSerialization.data(withJSONObject: dict),
```

Comment:
🤔 

We could revisit those `consoleDescription` to avoid using `Set`s. In this particular case, it'd be probably even better for the user if we always sorted the `Set`s. Or even don't use them at all.